### PR TITLE
SONAR-8460 Fix error message without args

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/setting/ws/SetAction.java [hunk 1]
```diff
@@ -288,7 +288,7 @@ private static SetRequest toWsRequest(Request request) {
     try {
       return (Map<String, String>) gson.fromJson(json, type);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/setting/ws/SetAction.java]: MINOR Remove this unnecessary cast to "Map". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/util/BooleanTypeValidation.java [hunk 2]
```diff
@@ -36,9 +36,8 @@ public String key() {
 
   @Override
   public void validate(String value, @Nullable List<String> options) {
-    if (!StringUtils.equalsIgnoreCase(value, "true") && !StringUtils.equalsIgnoreCase(value, "false")) {
-      throw new BadRequestException(format("Value '%s' must be one of \"true\" or \"false\".", value));
-    }
+    checkRequest(StringUtils.equalsIgnoreCase(value, "true") || StringUtils.equalsIgnoreCase(value, "false"),
+      format("Value '%s' must be one of \"true\" or \"false\".", value));
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/util/BooleanTypeValidation.java]: get rid of String.format

File: server/sonar-server/src/main/java/org/sonar/server/permission/ApplyPermissionTemplateQuery.java [hunk 3]
```diff
@@ -48,11 +48,7 @@ public String getTemplateUuid() {
   }
 
   private void validate() {
-    if (templateUuid == null || WHITESPACE.trimFrom(templateUuid).isEmpty()) {
-      throw new BadRequestException("Permission template is mandatory");
-    }
-    if (componentKeys == null || componentKeys.isEmpty()) {
-      throw new BadRequestException("No project provided. Please provide at least one project.");
-    }
+    checkRequest(templateUuid != null && !WHITESPACE.trimFrom(templateUuid).isEmpty(), "Permission template is mandatory");
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ApplyPermissionTemplateQuery.java]: Are you sure it will have the same behavior ?

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ApplyPermissionTemplateQuery.java]: exactly the same? But more consistent with how to check for blank string? yes

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ApplyPermissionTemplateQuery.java]: ok

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ExportAction.java [hunk 4]
```diff
@@ -119,10 +125,11 @@ public void handle(Request request, Response response) throws Exception {
     }
   }
 
-  private QualityProfileDto getProfile(DbSession dbSession, @Nullable String name, String language) {
-    QualityProfileDto profile = name == null ? dbClient.qualityProfileDao().selectDefaultProfile(dbSession, language)
-      : dbClient.qualityProfileDao().selectByNameAndLanguage(name, language, dbSession);
-    return checkFound(profile, "Could not find profile with name '%s' for language '%s'", name, language);
+  private QualityProfileDto getProfile(@Nullable String name, String language) {
+    try (DbSession dbSession = dbClient.openSession(false);) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ExportAction.java]: MINOR Remove this extraneous semicolon. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2959)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**